UnityProject/Assets/Scripts/Player/PlayerSync.cs

```diff
@@ -448,7 +454,9 @@ private void UpdateMe()
 			if (PlayerManager.MovementControllable == this as IPlayerControllable)
 			{
 				didWiggle = false;
-				if (KeyboardInputManager.IsMovementPressed() && Validations.CanInteract(playerScript,
+				bool inputDetected = KeyboardInputManager.IsMovementPressed();
+				CmdSetMovementInputState(inputDetected);
```

Comment:
```suggestion
				if(inputMovementDetected != inputDetected)
				{
					inputMovementDetected = inputDetected;
					CmdSetMovementInputState(inputDetected);
				}
```

So we don't spam the server

Reply:
but then the input will always be true because there's nothing telling the server that it's no longer true

Reply:
I made it so that the server sets it to false on each update but now the movement is incredibly clunky and random.

Reply:
![image](https://user-images.githubusercontent.com/34368774/150678216-44f351be-39ab-4162-8680-a1a4bb5f9f95.png)

Reply:
ok it's less clunky now that is 0.1f update but eh
I feel like this is going to be much worse on an actual server with players who have high ping because of all the delay that is being put ontop of each other right now.

either way this is good for now.

Reply:
> but then the input will always be true because there's nothing telling the server that it's no longer true

I was just reusing inputMovementDetected bool for the client (instead of making a new bool) to not send if it was already sent the same value.

The server would know the value as when the client stops pressing it will send false.